_posts/2021-10-05-str-methods.md

```diff
@@ -106,25 +173,25 @@ Welcome to Share to Learn
 Welcome to Learn to Share
 ```
 
+<div id="format_map"/>
+
 **format_map()**	Formats specified values in a string using a dictionary
 ```python
->>> lunch = {"Food":
-"Pizza", "Drink": "Wine"}
->>> print("Lunch: {Food},
-{Drink}".format_map(lunch))
-Lunch: Pizza, Wine
+>>> lunch = {
+            "Food": "Pizza", 
+            "Drink": "Juice"}
+>>> print("Lunch: {Food}, {Drink}".format_map(lunch))
+Lunch: Pizza, Juice
 >>> class Default(dict):
-def __missing__(self,
-key):
-return key
->>> lunch = {"Drink":
-"Wine"}
->>> print("Lunch: {Food},
-{Drink}".format_map(Default(
-lunch)))
-Lunch: Food, Wine
+        def __missing__(self, key):
```

Comment:
You can add a comment `# Handling the KeyError in case key is Not present in the dictionary.` for the readers who are coming across `__missing__`  dunder method for the first time.

Reply:
Done